### PR TITLE
Update buildtools to 2.0.0-prerelease-01714-04

### PR DESCRIPTION
**Pull request by chcosta**
Contains build fix for 

```
2017-06-14T13:39:20.2232748Z E:\A\_work\416\s\Tools\SyncCloudContent.targets(25,5): error : startIndex cannot be larger than length of string. [E:\A\_work\416\s\src\syncAzure.proj]
2017-06-14T13:39:20.2232748Z E:\A\_work\416\s\Tools\SyncCloudContent.targets(25,5): error : Parameter name: startIndex [E:\A\_work\416\s\src\syncAzure.proj]
2017-06-14T13:39:20.2232748Z E:\A\_work\416\s\Tools\SyncCloudContent.targets(25,5): error :    at System.String.Substring(Int32 startIndex, Int32 length) [E:\A\_work\416\s\src\syncAzure.proj]
2017-06-14T13:39:20.2232748Z E:\A\_work\416\s\Tools\SyncCloudContent.targets(25,5): error :    at System.String.Substring(Int32 startIndex) [E:\A\_work\416\s\src\syncAzure.proj]
2017-06-14T13:39:20.2232748Z E:\A\_work\416\s\Tools\SyncCloudContent.targets(25,5): error :    at Microsoft.DotNet.Build.CloudTestTasks.DownloadFromAzure.<ExecuteAsync>d__17.MoveNext() [E:\A\_work\416\s\src\syncAzure.proj]
2017-06-14T13:39:20.2388700Z Command execution failed with exit code 1.
2017-06-14T13:39:20.2388700Z ##[error]Process completed with exit code 1.
2017-06-14T13:39:20.2544948Z ##[section]Finishing: Sync packages
```

/cc @danmosemsft @gkhanna79

### CODE REVIEW

File: BuildToolsVersion.txt [hunk 1]
```diff
@@ -1 +1 @@
-2.0.0-prerelease-01708-02
\ No newline at end of file
+2.0.0-prerelease-01714-04
```
Comment (gkhanna79) [inline BuildToolsVersion.txt]: Please elaborate on the fix :)

Comment (chcosta) [inline BuildToolsVersion.txt]: https://github.com/dotnet/buildtools/pull/1572

CoreClr is the first repo to consume BuildTools after a recent refactoring and is the only repo making use of BlobNamePrefix to trim down to a specific filename prefix (other repos are either not using BlobNamePrefix or use it to trim to a specific folder within a container).  The fix made to buildtools is to ensure that we trim folder structure on a folder level, and still maintain compatibility with how other repo's are using the task.  

I've validated this locally for common coreclr, corefx, and core-setup usage of the downloadfromazure task.

Comment (gkhanna79) [inline BuildToolsVersion.txt]: Thanks - is this issue only in master?

CC @wtgodbe

Comment (chcosta) [inline BuildToolsVersion.txt]: I believe this is the only actionable issue.  If there are other issues, then they are hiding behind this one.  I noticed another failure, but it appeared to be network related, or a similar infrastructure problem.  Next, I'm going to start looking at the cleanup task to determine if we have a problem there.

**Comment by danmosemsft** [comment]
I'm merging because this is going into master and I've hit it.

**Comment by chcosta** [comment]
Thanks, I'll check on builds later tonight

**Pull request merged**